modules/tensor_mechanics/src/kernels/lagrangian/LagrangianStressDivergenceBase.C

```diff
@@ -0,0 +1,140 @@
+#include "LagrangianStressDivergenceBase.h"
+
+InputParameters
+LagrangianStressDivergenceBase::validParams()
+{
+  InputParameters params = Kernel::validParams();
+
+  params.addRequiredParam<unsigned int>("component", "Which direction this kernel acts in");
+  params.addRequiredCoupledVar("displacements", "The displacement components");
+
+  params.addParam<bool>("large_kinematics", false, "Use large displacement kinematics");
+  params.addParam<bool>("stabilize_strain", false, "Average the volumetric strains");
+
+  params.addParam<std::string>("base_name", "Material property base name");
+
+  params.addRequiredRangeCheckedParam<unsigned int>("component",
+                                                    "component < 3",
+                                                    "An integer corresponding to the direction "
+                                                    "the variable this kernel acts in. (0 for x, "
+                                                    "1 for y, 2 for z)");
+  params.addCoupledVar("temperature",
+                       "The name of the temperature variable used in the "
+                       "ComputeThermalExpansionEigenstrain.  (Not required for "
+                       "simulations without temperature coupling.)");
+
+  params.addParam<std::vector<MaterialPropertyName>>(
+      "eigenstrain_names",
+      "List of eigenstrains used in the strain calculation. Used for computing their derivatives "
+      "for off-diagonal Jacobian terms.");
+
+  return params;
+}
+
+LagrangianStressDivergenceBase::LagrangianStressDivergenceBase(const InputParameters & parameters)
+  : JvarMapKernelInterface<DerivativeMaterialInterface<Kernel>>(parameters),
+    _large_kinematics(getParam<bool>("large_kinematics")),
+    _stabilize_strain(getParam<bool>("stabilize_strain")),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _component(getParam<unsigned int>("component")),
+    _ndisp(coupledComponents("displacements")),
+    _disp_nums(_ndisp)
+{
+  // Do the vector coupling of the displacements
+  for (unsigned int i = 0; i < _ndisp; i++)
+    _disp_nums[i] = coupled("displacements", i);
+
+  // We need to use identical discretizations for all displacement components
+  auto order_x = getVar("displacements", 0)->order();
+  for (unsigned int i = 1; i < _ndisp; i++)
+  {
+    if (getVar("displacements", i)->order() != order_x)
+      mooseError("The Lagrangian StressDivergence kernels require equal "
+                 "order interpolation for all displacements.");
+  }
+
+  // Get the temperature, if coupled
+  if (isCoupled("temperature"))
+    _temperature = getVar("temperature", 0);
```

Comment:
It's better to move this to the initialization list and make the pointer const. For example
```cpp
_temperature(isCoupled("temperature") ? getVar("temperature", 0) : nullptr),
```

Reply:
Thanks, I fixed it.